src/python/pants/backend/codegen/thrift/java/thrift_defaults.py

```diff
@@ -25,12 +25,21 @@ def register_options(cls, register):
              help='The default language to generate for java_thrift_library targets.')
     register('--rpc-style', type=str, advanced=True, default='sync',
              help='The default rpc-style to generate for java_thrift_library targets.')
+    register('--default-java-namespace', type=str, advanced=True, default=None,
+             help='The default Java namespace to generate for java_thrift_library targets.')
+    register('--namespace-map', type=dict, advanced=True, default={},
+             help='The default namespace map to generate for java_thrift_library targets, {old: new}.')
+    register('--include-paths', type=list, advanced=True, default=[],
```

Comment:
Can you add a bit more information about how this would be used? We would generally prefer that people explicitly declare the dependencies of a target, which provides enough information to infer the include paths automatically.

So on the surface, I don't think this option makes sense: it would bypass change detection and allow a target to depend on an entire other subdirectory without declaring it as a dependency. In essence, it would exacerbate the internal ticket we have about this: `DPB-5639`, which states:
```
Currently, scrooge gen adds the source root of targets to the import paths of the global scrooge gen. This means that a target that depends on a thrift file [under a particular source root] can include any thrift [under that source root].

When there is an include in a thrift file whose target doesn't have matching dependency listings, incompatible changes to the included file can pass SQ, but will cause subsequent builds to fail.
```